sound/soc/sof/intel/hda-loader-skl.c

```diff
@@ -412,13 +411,6 @@ cl_skl_cldma_copy_to_buf(struct snd_sof_dev *sdev, const void *bin,
 			cl_skl_cldma_fill_buffer(sdev, bufsize, bufsize,
 						 curr_pos, true, true);
 
-			if (ret < 0) {
-				dev_err(sdev->dev, "error: fw failed to load. 0x%x bytes remaining\n",
-					bytes_left);
-				cl_skl_cldma_stream_run(sdev, false);
-				return ret;
-			}
-
```

Comment:
can you please rework the cl_skl_cldma_fill_buffer function? You don't need to pass two arguments for the buffer size and copy size.

Reply:
@plbossart As state as above, if using smaller DMA hardware buffer. we need copy multi times.
And the last buffer copy in may be smaller than the DMA hardware buffer, so we need buffer size and copy size(if size left is bigger than the DMA buffer, copy size equals to DMA buffer size, else using the left size). CSME will verify the buffer the same time as it copy in. And will run the firmware if the verify is OK.